### PR TITLE
[WCM] Serializer DateTimeNormalizer 

### PR DESCRIPTION
**Pull request by crayner**
Refers to [#31256](https://github.com/symfony/symfony/pull/31256)

Waiting Code Merge

### CODE REVIEW

File: components/serializer.rst [hunk 1]
```diff
@@ -721,6 +721,15 @@ There are several types of normalizers available:
     :phpclass:`DateTime` and :phpclass:`DateTimeImmutable`) into strings.
     By default it uses the RFC3339_ format.
 
+    The normaliser attempts to denormalize all time strings with a default timezone
+    if the defaultContext sets a `TIMEZONE_KEY` and `FORCE_CONTEXT_TIMEZONE` = true.
+    PHP :phpclass:`DateTimeInterface` constructor will return a :phpclass:`DateTimeInterface`
+    with a timezone that may not be the default timezone if the time-string has timezone
+    information embedded or the time-string is a UNIX timestamp representation.
+
+    A setting of `FORCE_CONTEXT_TIMEZONE` = false is considered deprecated from Symfony 4.2.
```
Comment (OskarStark) [inline components/serializer.rst]: Please add „.. deprecated::“ before

File: components/serializer.rst [hunk 2]
```diff
@@ -721,6 +721,15 @@ There are several types of normalizers available:
     :phpclass:`DateTime` and :phpclass:`DateTimeImmutable`) into strings.
     By default it uses the RFC3339_ format.
 
+    The normaliser attempts to denormalize all time strings with a default timezone
+    if the defaultContext sets a `TIMEZONE_KEY` and `FORCE_CONTEXT_TIMEZONE` = true.
```
Comment (OskarStark) [inline components/serializer.rst]: Please use double backticks for all highlightings

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: components/serializer.rst (hunk 1 above)
Comment (crayner) [reply components/serializer.rst]: Not sure as to what you wished enclosed here, so let me know if this is what is expected.

Are you aware of an editor that will show the results correctly.  Tis is my first time in the .rst world.

File: components/serializer.rst (hunk 2 above)
Comment (crayner) [reply components/serializer.rst]: Done

File: components/serializer.rst [hunk 3]
```diff
@@ -721,6 +721,15 @@ There are several types of normalizers available:
     :phpclass:`DateTime` and :phpclass:`DateTimeImmutable`) into strings.
     By default it uses the RFC3339_ format.
 
+    The normaliser attempts to denormalize all time strings with a default timezone
```
Comment (OskarStark) [inline components/serializer.rst]: s/normaliser/normalizer/

File: components/serializer.rst [hunk 4]
```diff
@@ -721,6 +721,15 @@ There are several types of normalizers available:
...
+    if the defaultContext sets a ``TIMEZONE_KEY`` and ``FORCE_CONTEXT_TIMEZONE`` = true.
```
Comment (OskarStark) [inline components/serializer.rst]: s/= true/is set to true/

Please put true, false and null in double backticks

I am currently on a phone 📱

File: components/serializer.rst [hunk 5]
```diff
@@ -721,6 +721,15 @@ There are several types of normalizers available:
...
+    PHP :phpclass:`DateTimeInterface` constructor will return a result with a timezone
+    that may not be the default timezone if the time-string has timezone information
```
Comment (OskarStark) [inline components/serializer.rst]: s/time-string/time string/

Same in the next line

File: components/serializer.rst [hunk 6]
```diff
@@ -721,6 +721,15 @@ There are several types of normalizers available:
...
+    embedded or the time-string is a UNIX timestamp representation.
+
+    A setting of ``FORCE_CONTEXT_TIMEZONE`` = false is considered ``.. deprecated::deprecated from Symfony 4.2``.
```
Comment (OskarStark) [inline components/serializer.rst]: .. deprecated:: 4.2

    Setting .... to ... is deprecated since Symfony 4.2. To use....

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: components/serializer.rst (hunk 3 above)
Comment (crayner) [reply components/serializer.rst]: Thanks for the feedback

File: components/serializer.rst (hunk 4 above)
Comment (crayner) [reply components/serializer.rst]: thanks for the feedback

File: components/serializer.rst (hunk 6 above)
Comment (crayner) [reply components/serializer.rst]: and again, thanls

File: components/serializer.rst [hunk 7]
```diff
@@ -721,6 +721,18 @@ There are several types of normalizers available:
...
+    The normalizer attempts to denormalize all time strings with a default timezone
+    if the defaultContext sets a ``TIMEZONE_KEY`` and ``FORCE_CONTEXT_TIMEZONE`` is
+    set to ``true``.
+
+    PHP :phpclass:`DateTimeInterface` constructor will return a result with a timezone
+    that may not be the default timezone if the time parameter has timezone information
+    embedded (e.g. 2010-01-28T15:00:00+02:00) or the time parameter is a UNIX timestamp
+    (e.g. @946684800). To use the default PHP behavior, set ``TIMEZONE_KEY`` to ``null``.
+
+    Setting ``FORCE_CONTEXT_TIMEZONE`` to ``false`` is considered deprecated from
```
Comment (OskarStark) [inline components/serializer.rst]:
```suggestion
    .. deprecated:: 4.2
    
        Setting ``FORCE_CONTEXT_TIMEZONE`` to ``false`` is considered deprecated from
        Symfony 4.2.
```

File: components/serializer.rst [hunk 8]
```diff
@@ -721,6 +721,18 @@ There are several types of normalizers available:
...
+    Symfony ``.. deprecated::4.2``.
```
Comment (OskarStark) [inline components/serializer.rst]:
```suggestion

```

**Review by OskarStark — CHANGES_REQUESTED** [review]
Sorry for beeing not so clear, I was on a phone.
Please apply my suggestions and we are good to go with the deprecation info.

Thanks.

File: components/serializer.rst (hunk 8 above)
Comment (crayner) [reply components/serializer.rst]: Good thinking

File: components/serializer.rst (hunk 7 above)
Comment (OskarStark) [reply components/serializer.rst]: Do have sind problems applying my suggestion?
We need the „.. deprecated:: 4.2“ directive

Comment (crayner) [reply components/serializer.rst]: Sorry, only my lack of understanding.  I can do that if you consider it important.

Comment (crayner) [reply components/serializer.rst]: Removed Deprecation completely.  No longer germane.

**Comment by javiereguiluz** [comment]
Closing because the related code PR was closed without merging.

**Pull request closed without merging**